pkg/storage/tsdb/users_scanner.go

```diff
@@ -65,12 +79,23 @@ func (s *UsersScanner) ScanUsers(ctx context.Context) (users, markedForDeletion
 			level.Warn(s.logger).Log("msg", "unable to check if user is marked for deletion", "user", userID, "err", err)
 		} else if deletionMarkExists {
 			users = append(users[:ix], users[ix+1:]...)
-			markedForDeletion = append(markedForDeletion, userID)
+			deletedUsers[userID] = struct{}{}
```

Comment:
Here we already did the IsOwned logic. Could we use a diff logic to avoid make it again?